pallets/stateful-storage/src/stateful_child_tree.rs

```diff
@@ -42,37 +243,50 @@ impl StatefulChildTree {
 	///
 	/// Allows getting all the keys having the same prefix
 	/// Warning: This should not be used from any on-chain transaction!
-	pub fn prefix_iterator<V: Decode + Sized>(
+	pub fn prefix_iterator<
+		V: Decode + Sized,
+		K: IsStatefulKey<H> + Sized,
+		PrefixKey: IsStatefulKey<H>,
+	>(
 		msa_id: &MessageSourceId,
-		keys: &[StatefulPageKeyPart],
-	) -> ChildTriePrefixIterator<(Vec<u8>, V)> {
+		prefix_keys: &PrefixKey,
+	) -> Box<impl Iterator<Item = (K, V)>> {
 		let child = Self::get_child_tree(*msa_id);
-		ChildTriePrefixIterator::<_>::with_prefix(&child, Self::concat_keys(keys).as_bytes_ref())
+		let result = ChildTriePrefixIterator::<(Vec<u8>, V)>::with_prefix(
+			&child,
+			&prefix_keys.hash_prefix_only(),
+		)
+		.filter_map(|(k, v)| {
+			if let Ok(key) =
```

Comment:
the match thing seems more of a rust way than an if statement. No strong opinion.

Reply:
It's more than an `if`... it's an `if let`. Which is just syntactic sugar for a basic `match`.